test/parse_test.dart

```diff
@@ -335,6 +335,25 @@ void main() {
         throwsFormat(parser, ['-mprofile']);
       });
 
+      test('throw if the value is not allowed by allowedHelp', () {
+        var parser = ArgParser();
+        parser.addOption('mode',
+            abbr: 'm',
+            allowedHelp: {'debug': 'debug mode', 'release': 'release mode'});
+
+        throwsFormat(parser, ['-mprofile']);
+      });
+
+      test('does not throw if value is allowed by allowedHelp', () {
+        var parser = ArgParser();
+        parser.addOption('mode',
+            abbr: 'm',
+            allowedHelp: {'debug': 'debug mode', 'release': 'release mode'});
+
+        // does not throw
+        parser.parse(['-mdebug']);
+      });
+
```

Comment:
Would also be good to test that passing both `allowed` and `allowedHelp` treats the former as the canonical list. So if `allowed` is `['a', 'b', 'c']` and `allowedHelp` is `{'a': 'A.', 'b': 'B.'}`, then you can still pass `c`.